src/components/[guild]/Requirements/components/RequirementImageEditor.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Box,
+  Button,
+  Circle,
+  FormControl,
+  FormErrorMessage,
+  Icon,
+  Img,
+  Text,
+  Wrap,
+} from "@chakra-ui/react"
+import usePinata from "hooks/usePinata"
+import { Upload, X } from "phosphor-react"
+import { useState } from "react"
+import { useDropzone } from "react-dropzone"
+import { useFormContext, useWatch } from "react-hook-form"
+import { useRequirementContext } from "./RequirementContext"
+
+const errorMessages = {
+  "file-too-large": "This image is too large, maximum allowed file size is 5MB",
+}
+
+const RequirementImageEditor = () => {
+  const { id } = useRequirementContext()
+
+  const { control, setValue } = useFormContext()
+  const requirements = useWatch({ name: "requirements", control })
+  const index = requirements.findIndex((requirement) => requirement.id === id)
+  const customImage = useWatch({
+    name: `requirements.${index}.data.customImage`,
+    control,
+  })
+
+  const [progress, setProgress] = useState<number>(0)
+
+  const uploader = usePinata({
+    onSuccess: ({ IpfsHash }) => {
+      setValue(
+        `requirements.${index}.data.customImage`,
+        `${process.env.NEXT_PUBLIC_IPFS_GATEWAY}${IpfsHash}`,
+        { shouldDirty: true }
+      )
+    },
+    onError: () => {
+      setValue(`requirements.${index}.data.customImage`, undefined)
+    },
+  })
+
+  const { fileRejections, getRootProps, getInputProps } = useDropzone({
+    multiple: false,
+    onDrop: (accepted) => {
+      if (accepted.length > 0) {
+        uploader.onUpload({ data: [accepted[0]], onProgress: setProgress })
+      }
+    },
+  })
+
+  return (
+    <FormControl isInvalid={!!fileRejections?.[0]}>
+      <Wrap>
+        {customImage ? (
+          <Box position={"relative"}>
+            <Circle
+              position="absolute"
+              onClick={() => {
+                setValue(`requirements.${index}.data.customImage`, undefined, {
```

Comment:
We might need to set this to empty string, but not sure about it, could you please test it?